webrtc/README.md

```diff
@@ -0,0 +1,445 @@
+least 1280
+bytes"](https://datatracker.ietf.org/doc/html/draft-ietf-quic-transport-29#section-14).
+An SCTP packet common header is 12 bytes long. An SCTP data chunk header size is
+16 bytes. Thus implementations SHOULD choose a message size equal or below 1252
+bytes. Long term we hope to be able to give better recommendations based on
+real-world experiments.
+
+### `RTCDataChannel` negotiation
+
+`RTCDataChannel`s are negotiated in-band by the WebRTC user agent (e.g. Firefox,
+Pion, ...). In other words libp2p WebRTC implementations MUST NOT change the
+default value `negotiated: false` when creating a `RTCDataChannel` via
+`RTCPeerConnection.createDataChannel`.
+
+The WebRTC user agent (i.e. not the application) decides on the `RTCDataChannel`
+ID based on the local node's connection role. For the interested reader see
+[RF8832 Protocol
+Overview](https://www.rfc-editor.org/rfc/rfc8832.html#section-4). It is
+RECOMMENDED that user agents reuse IDs once their `RTCDataChannel` closes. IDs
+MAY be reused according to RFC 8831: "Streams are available for reuse after a
+reset has been performed", see [RFC 8831 6.7 Closing a Data Channel
+](https://datatracker.ietf.org/doc/html/rfc8831#section-6.7). Up to 65535
+(`2^16`) concurrent data channels can be opened at any given time.
+
+According to RFC 8832 a `RTCDataChannel` initiator "MAY start sending messages
+containing user data without waiting for the reception of the corresponding
+DATA_CHANNEL_ACK message", thus using `negotiated: false` does not imply an
+additional round trip for each new `RTCDataChannel`.
+
+### `RTCDataChannel` label
+
+`RTCPeerConnection.createDataChannel()` requires passing a `label` for the
+to-be-created `RTCDataChannel`. When calling `createDataChannel` implementations
+MUST pass an empty string. When receiving an `RTCDataChannel` via
+`RTCPeerConnection.ondatachannel` implementations MUST NOT require `label` to be
+an empty string. This allows future versions of this specification to make use
+of the `RTCDataChannel` `label` property.
+
+## Connection Security
+
+Note that the below uses the message framing described in
+[multiplexing](#multiplexing).
+
+While WebRTC offers confidentiality and integrity via TLS, one still needs to
+authenticate the remote peer by its libp2p identity.
+
+After [Connection Establishment](#connection-establishment):
+
+1. _A_ and _B_ open a WebRTC data channel with `id: 0` and `negotiated: true`
+   ([`pc.createDataChannel("", {negotiated: true, id:
+   0});`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/createDataChannel)).
+
+2. _B_ starts a Noise `XX` handshake on the new channel. See
+   [noise-libp2p](https://github.com/libp2p/specs/tree/master/noise).
+
+   _A_ and _B_ use the [Noise
+   Prologue](https://noiseprotocol.org/noise.html#prologue) mechanism. More
+   specifically _A_ and _B_ set the Noise _Prologue_ to
+   `libp2p-webrtc-noise:<FINGERPRINT_A><FINGERPRINT_B>` before starting the
```

Comment:
Is the prologue a string or bytes? Because here you're concating a string with bytes.

Reply:
I used bytes when generating the test vectors.

Reply:
:+1:  Reworded in be5cfeb.